### PR TITLE
fix: add switch to open broadcast join when open window skew optimization

### PR DESCRIPTION
**Pull request by zekai427**
- Fix dataframe cache problem

### CODE REVIEW

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/WindowAggPlan.scala [hunk 1]
```diff
@@ -196,13 +192,18 @@ object WindowAggPlan {
       -1
     }
 
+    if (ctx.getConf.windowSkewOptCache) {
+      distributionDf = distributionDf.cache()
```
Comment (tobegit3hub) [inline java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/WindowAggPlan.scala]: Do not need to return and set the `distributionDf` object.

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/WindowAggPlan.scala [hunk 2]
```diff
@@ -196,13 +192,18 @@ object WindowAggPlan {
...
+    }
+
     // 2. Add "part" column and "expand" column by joining the distribution table
-    val addColumnsDf = SkewDataFrameUtils.genAddColumnsDf(inputDf, distributionDf, quantile.intValue(),
+    var addColumnsDf = SkewDataFrameUtils.genAddColumnsDf(inputDf, distributionDf, quantile.intValue(),
```
Comment (tobegit3hub) [inline java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/nodes/WindowAggPlan.scala]: Do not use `var` and set the `addColumnsDf` after calling `cache()`.

**Review by tobegit3hub — CHANGES_REQUESTED** [review]
This does not fix the issue of `broadcast`, we can link to another issue like unpersisting cache dataframes.

File: java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/utils/SkewDataFrameUtils.scala [hunk 3]
```diff
@@ -66,14 +66,20 @@ object SkewDataFrameUtils {
 
   def genAddColumnsDf(inputDf: DataFrame, distributionDf: DataFrame, quantile: Int,
                       repartitionColIndex: mutable.ArrayBuffer[Int], orderByColIndex: Int,
-                      partitionKeyColName: String, partIdColName: String, expandedRowColName: String): DataFrame = {
+                      partitionKeyColName: String, partIdColName: String, expandedRowColName: String,
+                      openBroadcastJoin: Boolean): DataFrame = {
```
Comment (tobegit3hub) [inline java/openmldb-batch/src/main/scala/com/_4paradigm/openmldb/batch/utils/SkewDataFrameUtils.scala:70]: You can pass the default value to set `openBroadcastJoin`.

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Review by dl239 — APPROVED** [review]

**Pull request merged**